RoddyCore/src/de/dkfz/roddy/plugins/LibrariesFactory.groovy

```diff
@@ -229,41 +238,64 @@ public class LibrariesFactory extends Initializable {
+                    continue;
+
                 String dirName = pEntry.getName();
-                boolean isZip = dirName.endsWith(".zip");
-                boolean unzippedDirectoryExists = false;
-                if (isZip) {
-                    dirName = dirName[0..-5]; // Remove .zip from the end.
-                    unzippedDirectoryExists = new File(dirName).exists();
-                    if (isZip && !unzippedDirectoryExists) warningUnzippedDirectoriesMissing = true;
-                    //set warn unzipped dir missing.
-                }
+                if (!isPluginDirectoryNameValid(dirName))
+                    continue;
 
+                if (!checkFile(new File(pEntry, "buildinfo.txt")) ||
+                        !checkFile(new File(pEntry, "buildversion.txt")) ||
+                        !checkDirectory(new File(pEntry, "resources/analysisTools")) ||
+                        !checkDirectory(new File(pEntry, "resources/configurationFiles"))
+                ) continue;
+
+//                boolean isZip = dirName.endsWith(".zip");
+//                boolean unzippedDirectoryExists = false;
+//                if (isZip) {
+//                    dirName = dirName[0..-5]; // Remove .zip from the end.
+//                    unzippedDirectoryExists = new File(dirName).exists();
+//                    if (isZip && !unzippedDirectoryExists) warningUnzippedDirectoriesMissing = true;
+//                    set warn unzipped dir missing.
+//                }
                 String[] splitName = dirName.split(StringConstants.SPLIT_UNDERSCORE); //First split for .zip then for the version
                 String pluginName = splitName[0];
-                if ((!pEntry.isDirectory() && !isZip) || isZip || !pluginName || blacklist.contains(pluginName))
-                    continue;
+//                if ((!pEntry.isDirectory() && !isZip) || isZip || !pluginName || blacklist.contains(pluginName))
+//                    continue;
                 collectedPluginDirectories << new Tuple2<File, String[]>(pEntry, splitName);
             }
         }
 
-        if (warningUnzippedDirectoriesMissing) {
-            logger.warning("There are plugins in your directories which are not unzipped. If some plugins are not found, please consider to check your zipped plugins.")
-        }
+//        if (warningUnzippedDirectoriesMissing) {
+//            logger.warning("There are plugins in your directories which are not unzipped. If some plugins are not found, please consider to check your zipped plugins.")
+//        }
 
         return loadPluginsFromDirectories(collectedPluginDirectories)
     }
 
     @groovy.transform.CompileStatic(TypeCheckingMode.SKIP)
-    private static List<Tuple2<File, String[]>> checkValidPluginNames (List<Tuple2<File, String[]>> collectedPluginDirectories) {
+    private static List<Tuple2<File, String[]>> checkValidPluginNames(List<Tuple2<File, String[]>> collectedPluginDirectories) {
```

Comment:
Not completely. A plugin name e.g. contains a ":" whereas a plugin dir contains "_" as a separator.